src/pfaffian.jl

```diff
@@ -0,0 +1,58 @@
+using LinearAlgebra: exactdiv
+
+# in-place O(n³) algorithm to compute the exact Pfaffian of
+# a skew-symmetric matrix over integers (or potentially any ring supporting exact division).
+#
+#     G. Galbiati & F. Maffioli, "On the computation of pfaffians,"
+#     Discrete Appl. Math. 51, 269–275 (1994).
+#     https://doi.org/10.1016/0166-218X(92)00034-J
+function exactpfaffian!(A::AbstractMatrix{<:Integer})
+    LinearAlgebra.require_one_based_indexing(A)
+    isskewhermitian(A) || throw(ArgumentError("Pfaffian requires a skew-Hermitian matrix"))
+    n = size(A,1)
+    isodd(n) && return zero(eltype(A))
+    c = one(eltype(A))
+    signflip = false
+    n = n ÷ 2
+    while n > 1
+        # find last k with A[2n-1,k] ≠ 0
+        k = 2n
+        while k > 0 && iszero(A[2n-1,k]); k -= 1; end
+        iszero(k) && return zero(eltype(A))
+        
+        # swap rows/cols k and 2n
+        if k != 2n
+            for i = 1:2n
+                A[i,k], A[i,2n] = A[i,2n], A[i,k]
+                A[k,i], A[2n,i] = A[2n,i], A[k,i]
+            end
+            signflip = !signflip
+        end
+        
+        # update, A, c, n
+        for j = 1:2n-2, i = 1:j-1
+            δ = A[2n-1,2n]*A[i,j] - A[i,2n-1]*A[j,2n] + A[j,2n-1]*A[i,2n]
+            A[j,i] = -(A[i,j] = exactdiv(δ, c))
```

Comment:
Just use `div` since we restricted this to integers.

(The `exactdiv` function was introduced in Julia 1.7 for https://github.com/JuliaLang/julia/pull/40868, which added an exact determinant algorithm for arbitrary rings via the Bareiss algorithm.  However, I'm not 100% sure that the algorithm here works for arbitrary rings, and in particular it might implicitly assume that `*` commutative, so it is safer to restrict it to `<:Integer`, at least for now.   In this case `exactdiv == div`.)